test/unit/math/prim/fun/quantile_test.cpp

```diff
@@ -0,0 +1,134 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+
+template <typename T>
+void test_quantile_double() {
+  using stan::math::index_type_t;
+  using stan::math::quantile;
+
+  T c0(0);
+  EXPECT_THROW(quantile(c0, 0.3), std::invalid_argument);
+
+  T c(1);
+  c[0] = 1.7;
+  EXPECT_EQ(quantile(c, 0), 1.7);
+  EXPECT_EQ(quantile(c, 1), 1.7);
+  EXPECT_EQ(quantile(c, 0.33), 1.7);
+  EXPECT_EQ(quantile(c, 0.68), 1.7);
+
```

Comment:
Check that the function throws if the second argument is outsize [0, 1].

Also check nans throw.

Reply:
thanks, added